tests/zexperimental/test_mcmc.py

```diff
@@ -265,6 +272,35 @@ def test_CWMH_regression_warmup(target: cuqi.density.Density):
                                         old_idx=[0, -1],
                                         new_idx=[1, None])
 
+# ============= HMC (NUTS) ==============
```

Comment:
After merging of https://github.com/CUQI-DTU/CUQIpy/pull/390 there may be some other tests to add the NUTS sampler in. Consider rebasing and adding those.

Reply:
The main was merged into this branch. I added NUTS to the skip tests for checkpointing. My plan is to address checkpointing in #404